pkg/fseval/mknod_freebsd.go

```diff
@@ -0,0 +1,28 @@
+/*
+ * umoci: Umoci Modifies Open Containers' Images
+ * Copyright (C) 2016-2020 SUSE LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fseval
+
+import (
+	"os"
+
+	"golang.org/x/sys/unix"
+)
+
+func (fs osFsEval) Mknod(path string, mode os.FileMode, dev uint64) error {
+       return unix.Mknod(path, uint32(mode), uint64(dev))
```

Comment:
The right place to do this split IMHO is in `pkg/system/` by making `mknod_linux.go` and `mknod_bsd.go` files which wrap `unix.Mknod`. Then swap all usages of `unix.Mknod` with `system.Mknod`. That way you only end up with two OS-specific files in a directory that already contains them, rather than having four spread out inside `pkg`.

Reply:
@cyphar I think I addressed your feedback. Please take a look.